### ISSUE TITLE
It's not necessary to sum up 'log_probs' at the last dimension in discretized_mix_logistic_loss function

### ISSUE THREAD

**Issue by naturomics** [r9y9/wavenet_vocoder#33]
Hi Ryuichi,

Thanks for your great work.

Forgive me if I got this wrong but I think it's not necessary to sum up `log_probs`  at the last dimension [here](https://github.com/r9y9/wavenet_vocoder/blob/2bf9e78fdee5aef16a63747c82691877fa70c413/wavenet_vocoder/mixture.py#L101). Saying
> log_probs = torch.sum(log_probs, dim=-1, keepdim=True) + F.log_softmax(logit_probs, -1)

should be
> log_probs = log_probs + F.log_softmax(logit_probs, -1)

I saw your code is adapted from [pixel-cnn](https://github.com/openai/pixel-cnn), but these two cases are different. Since pixel has 3 channels(RGB) in pixelCNN and the `log_probs` has shape [B, H, W, C, num_mix] before this [line](https://github.com/openai/pixel-cnn/blob/53ad661b8fa12e926286f0e48f5fbe0c8c874532/pixel_cnn_pp/nn.py#L83), the mixture model is built on pixel not channel so they get the probability of each pixel with `tf.reduce_sum(log_probs,axis=3)`. But here for the wavenet, waveform has only one channel, saying [`log_probs`](https://github.com/r9y9/wavenet_vocoder/blob/2bf9e78fdee5aef16a63747c82691877fa70c413/wavenet_vocoder/mixture.py#L99) has shape [B, T, num_mix], and the mixture model is built on that channel, so it's not needed to sum up at any dimension. In addition, your code summing up at the `num_mix`  dimension is also inconsistent with pixel cnn(which at channels dimension).

Feel free to ignore me if I'm wrong,
Thanks

**Comment by jiqizaisikao** [comment]
yes,i have also find the same problem

**Comment by jiqizaisikao** [comment]
I have one question:the integration of p from (-1,1) should be 1,but it seems that there is no guarantee to it。Like the original code the p is so small like exp(-50)  and  integration of p will not be 1.I have trained 320k iters,and I have use  my modified version the p is like exp(-5)

**Comment by r9y9** [comment]
Thank you for the comment. I would like to think about it again but do not have time currently:( If you are sure it's a right fix and can get good speech quality, that would be great. 

From https://github.com/r9y9/wavenet_vocoder/issues/1#issuecomment-359735902:

> Honestly, I am not completely sure if I implemented it correctly, so I regard the feature as experimental at the moment. Let me know if you find a bug. PRs are always welcome!

**Comment by r9y9** [comment]
@naturomics Sorry for super late reply. I think you are right. Thank you for pointing this out! Much appreciated.
